fix: use only positive angles, add test for this

haval0 opened the pull request:
resolves #120

the angle function in utils has been changed to return the absolute value of its result, and a test has been added, and another test fixed, to make sure of this. All abs() usages now return double instead of int.

haval0 commented:
This is meant to ensure that "The tests should break if someone changes anything important", like our canvas feedback states

haval0 commented on decide/utils.cpp:
```diff
@@ -12,7 +14,7 @@ double angle_between_points(Coordinate A, Coordinate B, Coordinate C) {
   double a = atan2(B.y - A.y, B.x - A.x);
   double b = atan2(C.y - A.y, C.x - A.x);
-  return b - a;
+  return abs(b - a);
```
Alternatively, instead of using `abs()`, this could be solved by adding `pi`. I don't think it makes a difference for this program.

danielwis replied:
Ah, I see. I think `abs` is clearer since we don't care which line represents the x-axis (if we think of a negative angle as extending below the x-axis, that is), but even if we were to change it, shouldn't it be `2*pi`?

haval0 replied:
You are right, it would be adding 2pi as you say, but only if the result of `b - a` is negative. otherwise do nothing. But yes let's leave it as is

haval0 commented:
- [ ] Feature/fix PRs should add one atomic (as small as possible) feature or
  fix
- [ ] If your PR adds a feature or fix it also needs to add or modify at least
  one test
- [ ] All code in your PR needs to have been formatted by `clang-format`
- [ ] The merge commit title should follow our commit prefix convention of
  either "feat", "fix", "doc", or "refactor"
- [ ] The merge commit body should reference at least one issue

danielwis commented on decide/utils.test.cpp:
```diff
@@ -8,12 +8,27 @@ void test_angle_90_degree() {
+  double angle3 = angle_between_points(Coordinate(-1, -1), Coordinate(0, 0),
+                                       Coordinate(1, 0));
+
+  assert((0 <= angle1) && (angle1 < 2 * pi));
```
Should we perhaps test their numerical values instead of just testing that the values given are allowed?

haval0 replied:
That would indeed be even more useful.  Will see if I can add that later.

haval0 commented:
Caught a bad bug where all our `abs()` usages were converting to `int`. Changed all occurrences in the code base to `std::abs()` so that `cmath`'s overload that returns a double is now used.

Also fixed all the angle tests to be more correct and thorough.

haval0 commented:
Caught a bad bug where all our `abs()` usages were converting to `int`. Changed all occurrences in the code base to `std::abs()` so that `cmath`'s overload that returns a double is now used.

Also fixed all the angle tests to be more correct and thorough.

danielwis reviewed (approved):
LGTM now. Good job!
